File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/config/impl/EnemyConfig.java
package com.dkellycollins.triggerfinger.data.config.impl;

import android.content.Context;

import com.dkellycollins.triggerfinger.R;
import com.dkellycollins.triggerfinger.data.config.IEnemyConfig;

public class EnemyConfig extends RConfig implements IEnemyConfig {

    public EnemyConfig(Context context) {
        super(context);
    }

    @Override
    public int getRespawnRate() {
        return _context.getResources().getInteger(R.integer.enemy_respawn_rate);
    }

    @Override
    public float getCollidableRadius() {
        return getFloat(R.integer.enemy_collidable_radius);
    }

    @Override
    public float getDeltaY() {
        return getFloat(R.integer.enemy_delta_y);
    }
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/config/impl/EnemyConfig.kt
package com.dkellycollins.triggerfinger.data.config.impl

import android.content.Context

import com.dkellycollins.triggerfinger.R
import com.dkellycollins.triggerfinger.data.config.IEnemyConfig

class EnemyConfig(context: Context) : RConfig(context), IEnemyConfig {

    override val spriteId: Int
        get() = R.drawable.enemy_sprite

    override val respawnRate: Int
        get() = _context.resources.getInteger(R.integer.enemy_respawn_rate)

    override val collidableRadius: Float
        get() = getFloat(R.integer.enemy_collidable_radius)

    override val deltaY: Float
        get() = getFloat(R.integer.enemy_delta_y)
}
